### PR TITLE
refactor: move Presention Definition validation to custom validator

### PR DESCRIPTION
**Pull request by jrhender**
Main changes are:
- Add `IsPresentationDefinitionCredentialQuery` custom validator (which uses `@sphereon/pex`)
- Fix nested validation in `exchange-definition.dto` (needed to add `@Type`)

### CODE REVIEW

File: apps/nestjs-wallet/src/vc-api/exchanges/dtos/custom-validators/presentation-definition-credential-query.validator.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      constraints: [],
+      options: validationOptions,
+      validator: {
+        validate(value: IPresentationDefinition, args: ValidationArguments) {
+          const pex = new PEX();
+          const validated = pex.validateDefinition(value);
+          if (Array.isArray(validated)) {
+            validated.forEach((checked) => {
+              if (checked.status === Status.ERROR || checked.status === Status.WARN) {
+                return false;
```
Comment (Harasz) [inline apps/nestjs-wallet/src/vc-api/exchanges/dtos/custom-validators/presentation-definition-credential-query.validator.ts:22]: I don’t think it’s gone work as you wished. This is a return statements for a ‘forEach’ method. It looks like for me that’s array will always be valid object in this case.

Comment (jrhender) [reply apps/nestjs-wallet/src/vc-api/exchanges/dtos/custom-validators/presentation-definition-credential-query.validator.ts]: Lol, nice catch, thanks 🤦

File: apps/nestjs-wallet/src/vc-api/exchanges/dtos/custom-validators/presentation-definition-credential-query.validator.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      options: validationOptions,
+      validator: {
+        validate(value: IPresentationDefinition, args: ValidationArguments) {
+          const pex = new PEX();
+          try {
+            const validated = pex.validateDefinition(value);
+            const resultArray = Array.isArray(validated) ? validated : [validated];
+            const statuses = resultArray.map((checked) => checked.status);
+            if (statuses.includes(Status.ERROR) || statuses.includes(Status.WARN)) {
+              return false;
```
Comment (Harasz) [inline apps/nestjs-wallet/src/vc-api/exchanges/dtos/custom-validators/presentation-definition-credential-query.validator.ts:23]: What do you think about throwing here an error? With this approach you will have only one return statements in `catch` block. Also you have only one place for "invalidating" logic in your code

**Review by Harasz — APPROVED** [review]

File: apps/nestjs-wallet/src/vc-api/exchanges/dtos/custom-validators/presentation-definition-credential-query.validator.ts (hunk 2 above)
Comment (jrhender) [reply apps/nestjs-wallet/src/vc-api/exchanges/dtos/custom-validators/presentation-definition-credential-query.validator.ts:23]: Hmm, maybe I don't understand the custom validators for `class-validator` that well. But shouldn't a validator return true or false? I think that though `pex` might throw an error to indicate an invalid presentation definition, I think it makes sense for the return of the custom validator to reliably be true or false. Is that not how you see it?

**Pull request merged**